UndertaleModLib/Models/UndertaleEmbeddedTexture.cs

```diff
@@ -169,16 +231,17 @@ public void Unserialize(UndertaleReader reader)
                     reader.Position += 8;
 
                     // Need to fully decompress and convert the QOI data to PNG for compatibility purposes (at least for now)
-                    using MemoryStream bufferWrapper = new MemoryStream(reader.Buffer);
-                    bufferWrapper.Seek(reader.Offset, SeekOrigin.Begin);
-                    using MemoryStream result = new MemoryStream(1024);
-                    BZip2.Decompress(bufferWrapper, result, false);
-                    reader.Position = (uint)bufferWrapper.Position;
-                    result.Seek(0, SeekOrigin.Begin);
-                    using Bitmap bmp = QoiConverter.GetImageFromSpan(result.GetBuffer());
-                    using MemoryStream final = new MemoryStream();
-                    bmp.Save(final, ImageFormat.Png);
-                    TextureBlob = final.ToArray();
+                    if (sharedStream.Length != 0)
+                        sharedStream.Seek(0, SeekOrigin.Begin);
+                    BZip2.Decompress(reader.Stream, sharedStream, false);
+                    using Bitmap bmp = QoiConverter.GetImageFromSpan(sharedStream.GetBuffer().AsSpan()[..(int)sharedStream.Position]);
```

Comment:
Can you create a local var for `sharedStream.GetBuffer().AsSpan()[..(int)sharedStream.Position]` for readibilty?